optimizer_wrapper.rb

```diff
@@ -98,22 +99,26 @@ def self.wrapper_vrp(api_key, services, vrp, checksum, job_id = nil)
 
     if services_vrps.any?{ |sv| !sv[:service] }
       raise UnsupportedProblemError.new('Cannot apply any of the solver services', inapplicable_services)
-    elsif vrp.restitution_geometry && !vrp.points.all?{ |point| point[:location] }
-      raise DiscordantProblemError, 'Geometry is not available if locations are not defined'
+    elsif vrp.restitution_geometry.any? && !vrp.points.all?(&:location)
+      raise DiscordantProblemError.new('Geometry is not available if locations are not defined')
+    elsif config[:solve][:synchronously] || (
+            services_vrps.size == 1 &&
+            !vrp.preprocessing_cluster_threshold &&
+            config[:services][services_vrps[0][:service]].solve_synchronous?(vrp)
+          )
+      # The job seems easy enough to perform it with the server
+      result = define_main_process(services_vrps, job_id)
+      result.size == 1 ? result.first : result
     else
-      if config[:solve][:synchronously] || (services_vrps.size == 1 && !vrp.preprocessing_cluster_threshold && config[:services][services_vrps[0][:service]].solve_synchronous?(vrp))
-        # The job seems easy enough to perform it with the server
-        result = define_main_process(services_vrps, job_id)
-        (result.size == 1) ? result.first : result
-      else
-        # Delegate the job to a worker
-        job_id = Job.enqueue_to(services[:queue], Job, services_vrps: Base64.encode64(Marshal.dump(services_vrps)),
-                                                       api_key: api_key,
-                                                       checksum: checksum,
-                                                       pids: [])
-        JobList.add(api_key, job_id)
-        job_id
-      end
+      # Delegate the job to a worker
+      job_id = Job.enqueue_to(services[:queue], Job, services_vrps: Base64.encode64(Marshal.dump(services_vrps)),
+                                                     api_key: api_key,
+                                                     checksum: checksum,
+                                                     pids: [],
+                                                     configuration: { csv: services_vrps.any?{ |s| s[:vrp].restitution_csv },
+                                                                      geometry: [] })
```

Comment:
Do we need this `configuration:` hash in job?

Reply:
One test failed without it, but it does not anymore. I can remove this eventually, test has probably been fixed when we modified call to OutputHelper::Result.generate_geometry(solution) which now handles correctly when solution is empty (and hence no configuration is provided).